src/front/glsl/parser/functions.rs

```diff
@@ -511,7 +511,17 @@ impl<'source> ParsingContext<'source> {
                 self.expect(parser, TokenValue::Semicolon)?.meta
             }
             TokenValue::Semicolon => self.bump(parser)?.meta,
-            _ => meta,
+            _ => {
+                let Token { value, meta } = self.bump(parser)?;
+                return Err(Error {
+                    kind: ErrorKind::InvalidToken(
+                        value,
+                        // TODO: Add all tokens that can start a statement?
+                        vec![TokenValue::RightBrace.into()],
+                    ),
+                    meta,
+                });
+            }
```

Comment:
We can just force a expression parsing if we don't recognized a token since they'll both result in a error, but we get the benefit that the error already has the expected tokens and we don't get cases like #1232 where the initial token wasn't recognized as valid for an expression.

Reply:
Sure. So just replace all the `TokenValue::Plus | TokenValue::Dash | TokenValue::Bang | ...` that currently forwards to expression parsing, collapsing it to simply `_ => { <attempt to parse expression> }`?

Reply:
Exactly